src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -416,17 +415,32 @@ def symlink_or_copy(
 
     # Verify that the mapping isn't accidentally trying to create a file in the project source through symlinks.
     # We need to ensure we're resolving symlinks for this check to be effective.
+    # We are unlikely to hit this if calling the function through bundle map, keeping it here for other future use cases outside bundle.
     resolved_dst = dst.resolve()
     resolved_deploy_root = deploy_root.resolve()
-    if resolved_deploy_root not in resolved_dst.parents:
+    dst_is_deploy_root = resolved_deploy_root == resolved_dst
+    if (not dst_is_deploy_root) and (resolved_deploy_root not in resolved_dst.parents):
         raise NotInDeployRootError(dest_path=dst, deploy_root=deploy_root, src_path=src)
 
-    if overwrite:
+    if (not dst_is_deploy_root) and overwrite:
+        # TODO Verify: We do not want to delete if dst_is_deploy_root as it may contain artifacts from other runs of bundle map
+        # But this will treat deploy_root as a special dst, as compared to another dst directory that may also have other artifacts from a prev run
         delete(dst)
-    try:
-        os.symlink(src, dst)
-    except OSError:
-        ssrc.copy(dst)
+
+    absolute_src = resolve_without_follow(src)
+    if absolute_src.is_file():
+        try:
+            os.symlink(absolute_src, dst)
+        except OSError:
+            ssrc.copy(dst)
+    else:
+        # 1. Create an empty shell directory in the deploy root
+        dst.mkdir(exist_ok=True)
+        # 2. Get all children of the current directory, files and sub-directories
+        for child in sorted(absolute_src.iterdir()):
+            child_path = Path(absolute_src, child.name)
+            dest_path = Path(dst, child.name)
+            symlink_or_copy(src=child_path, dst=dest_path, deploy_root=deploy_root)
```

Comment:
Is this vulnerable to infinite loops if a subdir is a symlink pointing back to an earlier dir? AFAIK os.walk handles that just fine, but I'm worried that this recursive logic wouldn't.